lib/ffi/dynamic_library.rb

```diff
@@ -0,0 +1,89 @@
+#
+# Copyright (C) 2008-2010 Wayne Meissner
+#
+# This file is part of ruby-ffi.
+#
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+# * Redistributions in binary form must reproduce the above copyright notice
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+# * Neither the name of the Ruby FFI project nor the names of its contributors
+#   may be used to endorse or promote products derived from this software
+#   without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.#
+
+module FFI
+  class DynamicLibrary
+    SEARCH_PATH = %w[/usr/lib /usr/local/lib /opt/local/lib]
+    if FFI::Platform::ARCH == 'aarch64' && FFI::Platform.mac?
+      SEARCH_PATH << '/opt/homebrew/lib'
+    end
```

Comment:
Not sure if this is something you want to handle (and I wouldn't blame you if you don't):

Some Homebrew users install `brew` to a path that is neither `/usr/local` nor `/opt/homebrew` (e.g. they have no `sudo` access so use their `$HOME`). For these cases, one approach that works is to check if `brew` is in `PATH` and then check the output of `brew --prefix`.

Reply:
That seems an orthogonal issue to me.
Shelling out to `brew --prefix` seems quite expensive, maybe there should be a standard `HOMEBREW_PREFIX` env var exported by default or so, so we could read it without the cost of a subprocess.
Could you file a separate issue if you need that? (or maybe there is already one?)

Reply:
@eregon `HOMEBREW_PREFIX` is set by `brew` but running `brew --prefix` is the correct way to detect the homebrew location. or we need a configure flag for prepending an include dir to the initial build of the gem so that people can set their own configurations appropriately. `/opt/homebrew` and `/usr/local` are standard locations, but not the only options

Reply:
Yes I agree this is not a blocker. I think `brew --prefix` is too much of a cost on `require`, but maybe it's OK to compute it once lazily on `ffi_lib` on macOS.

What we need is @larskanis to review this PR and then make a release.